src/System.Buffers.Primitives/System/Buffers/Transformation.cs

```diff
@@ -8,6 +8,10 @@ public interface IBufferOperation
     {
         OperationStatus Execute(ReadOnlySpan<byte> input, Span<byte> output, out int consumed, out int written);
     }
+    public interface IBufferTransformation
```

Comment:
Why do we need a separate interface? Can we add the in-place transform to IBufferOperation?

Reply:
@ahsonkhan Not all transforms would support in-place. IMO it makes sense to have a separate interface for this capability. (Though you'd also probably want a `CanTransformInPlace` API just in case you're subclassing a base class that implements this interface and you need to suppress this capability.)

Reply:
Some transformations cannot be done in-place